### ISSUE TITLE
Cannot send funds right after receiving them

### ISSUE THREAD

**Issue by nop33** [alephium/desktop-wallet#206]
**Describe the bug**
In my local devnet (node 1.3.2, expl-bck 1.6.0) I noticed that after I send a tx from address `X` to another `Y` and
once the explorer API `getAddressesAddressTransactions` endpoint (get confirmed txs) as well as the `getAddressesAddress` (get address details) include this tx, then I have an issue building a new transaction from address `Y`, unless I wait for a minute or so. It seems that the node is not up to date with the latest tx. It returns an error of “insufficient funds”. Then, after a few seconds/minutes, it stops returning this error.

**To Reproduce**

https://user-images.githubusercontent.com/1579899/163135605-2b71f8d3-a917-4b15-b39c-6d63538c07fa.mov

**Expected behavior**
Be able to send the funds once a confirmed tx is received

**Additional context**
It appears that we wait for `toGroupConfirmations >= 1` to be able to make a transaction from `Y`. This number is from `/transactions/status`:

```json
{
  "type": "Confirmed",
  "blockHash": "bdaf9dc514ce7d34b6474b8ca10a3dfb93ba997cb9d5ff1ea724ebe2af48abe5",
  "txIndex": 0,
  "chainConfirmations": 1,
  "fromGroupConfirmations": 2,
  "toGroupConfirmations": 3
}
```

If `Y` has funds in the wallet, `Y` could use the existing UTXOs to build transactions.  
If `Y` is a new address, it will need to wait a bit more. For now, we could probably mark the tx as pending if `toGroupConfirmations < 1` 

Related issue: https://github.com/alephium/explorer-backend/issues/198

**Comment by lf94** [comment]
> If Y has funds in the wallet, Y could use the existing UTXOs to build transactions.

This will be the case already right? If there were existing UTXOs to use, there'd be sufficient funds?

> If Y is a new address, it will need to wait a bit more. For now, we could probably mark the tx as pending if toGroupConfirmations < 1

Yep, we need to consider anything with less than 1 confirmations to be pending / excluded from the total funds.

**Comment by lf94** [comment]
I can't reproduce on testnet. I think it may have to do with your local devnet setup?

**Comment by lf94** [comment]
So the error is actually coming from the node... so the node can't build a TX with a zero confirmation UTXO then? Ok.

**Comment by lf94** [comment]
The more I think about this the more unsure I am of it...

Is this going to involve making a request for every single transaction to check its confirmation status?

**Comment by nop33** [comment]
I think it's a UI issue. I guess the wallet should not consider all transactions returned from the `addresses.getAddressesAddressTransactions` endpoint as confirmed, but only those with `toGroupConfirmations >= 1`. However, `toGroupConfirmations` is not available in the endpoint response...

**Comment by lf94** [comment]
Exactly, we have a need for more information in a transaction information API request. We can't be making an additional request for every single transaction to get their confirmations.

Mhm, it is a UI issue, sorry if it seemed like I was expressing something else; I meant that the node error we're showing when trying to submit the TX is coming from the node and not within the wallet itself.

/cc @tdroxler

**Comment by LeeAlephium** [comment]
reminder @tdroxler

**Comment by LeeAlephium** [comment]
There are currently a few immediate solutions I see for this issue:

* alephium/alephium can use zero confirmation UTXOs to build transactions
* alephium/explorer-backend offers the needed functionality
* alephium/desktop-wallet uses the same method to calculate the confirmations as explorer but it will generate tons of network activity
* alephium/desktop-wallet makes it clear the wallet total is an "estimate" and only calculates at the moment of transaction how much is available (confusing)
